File: cron.php
<?php

$MYSQL_HOST = "srv1423.hstgr.io";
$MYSQL_USER = "u431114049_exoplanets_adm";
$MYSQL_PASS = "4~YD]2Bg";
$MYSQL_DB = "u431114049_exoplanets";
$BASE_URL = 'https://exoplanetarchive.ipac.caltech.edu/TAP/sync?query=SELECT%20*%20FROM%20pscomppars&format=json';

function jsonToInserts($apiUrl, $tableName, $dbConfig) {
    try {
        // Definir las columnas permitidas
        $allowedColumns = [
            "objectid", "pl_name", "pl_letter", "pl_status", "pl_bmasse", "pl_orbper", "pl_orbeccen",
            "pl_radj", "pl_rade", "pl_massj", "pl_masse", "hostid", "hostname", "disc_pubdate",
            "disc_year", "disc_method", "discoverymethod", "disc_locale", "disc_facility",
            "disc_instrument", "disc_telescope", "disc_refname", "ra", "rasymerr", "rastr",
            "ra_solnid", "ra_reflink", "dec", "decsymerr", "decstr", "dec_solnid", "dec_reflink",
            "glon", "glonstr", "glon_solnid", "glon_reflink", "glat", "glatstr", "glat_solnid",
            "glat_reflink", "elon", "elonstr", "elon_solnid", "elon_reflink", "elat", "elat_solnid",
            "elat_reflink", "elatstr", "sy_snum", "sy_dist"
        ];

        // Llamada a la API
        $response = file_get_contents($apiUrl);
        if ($response === FALSE) {
            throw new Exception("Error al obtener datos de la API.");
        }

        // Convertir la respuesta a JSON
        $data = json_decode($response, true);
        if (!is_array($data)) {
            throw new Exception("El JSON recibido no es un array.");
        }

        // Conectar a la base de datos
        $conn = new mysqli($dbConfig['host'], $dbConfig['user'], $dbConfig['password'], $dbConfig['database']);
        if ($conn->connect_error) {
            throw new Exception("Error de conexión a la base de datos: " . $conn->connect_error);
        }

        // Eliminar datos existentes de la tabla
        $conn->query("DELETE FROM `" . $tableName . "`;");

        // Insertar datos
        foreach ($data as $record) {
            $filteredRecord = array_filter(
                $record,
                function($key) use ($allowedColumns) {
                    return in_array($key, $allowedColumns);
                },
                ARRAY_FILTER_USE_KEY
            );

            if (empty($filteredRecord)) continue;

            $columns = "`" . implode("`, `", array_keys($filteredRecord)) . "`";
            $values = array_map(function($value) use ($conn) {
                if ($value === null) return "NULL";
                return is_string($value) ? "'" . $conn->real_escape_string($value) . "'" : $value;
            }, array_values($filteredRecord));

            $valuesList = implode(", ", $values);
            $insertStatement = "INSERT INTO `$tableName` ($columns) VALUES ($valuesList);";

            if (!$conn->query($insertStatement)) {
                throw new Exception("Error al insertar datos: " . $conn->error);
            }
        }

        // Confirmar cambios
        $conn->commit();

        // Cerrar conexión
        $conn->close();

        echo "Datos insertados correctamente en la tabla $tableName.\n";

    } catch (Exception $e) {
        echo "Error: " . $e->getMessage() . "\n";
    }
}

// Ejemplo de uso
$apiUrl = $BASE_URL;
$tableName = "pscomppars";

$dbConfig = [
    'host' => $MYSQL_HOST,
    'user' => $MYSQL_USER,
    'password' => $MYSQL_PASS,
    'database' => $MYSQL_DB
];

jsonToInserts($apiUrl, $tableName, $dbConfig);

File: cron.py
import requests
import json
import mysql.connector

MYSQL_HOST = "srv1423.hstgr.io"
MYSQL_USER = "u431114049_exoplanets_adm"
MYSQL_PASS = "4~YD]2Bg"
MYSQL_DB = "u431114049_exoplanets"
BASE_URL = 'https://exoplanetarchive.ipac.caltech.edu/TAP/sync?query=SELECT%20*%20FROM%20pscomppars&format=json'

def json_to_inserts(api_url, table_name, db_config):
    try:
        # Llamada a la API
        response = requests.get(api_url)
        response.raise_for_status()

        # Convertir la respuesta a JSON
        data = response.json()

        # Asegurarse de que el JSON sea un array
        if not isinstance(data, list):
            raise ValueError("El JSON recibido no es un array.")

        # Detectar las columnas automáticamente
        if len(data) == 0:
            raise ValueError("El JSON está vacío y no se pueden detectar columnas.")

        column_names = list(data[0].keys())

        # Conectar a la base de datos
        conn = mysql.connector.connect(**db_config)
        cursor = conn.cursor()

        # Eliminar datos existentes de la tabla
        cursor.execute(f"DELETE FROM {table_name};")

        # Insertar datos
        for record in data:
            values = []
            for column in column_names:
                value = record.get(column, None)
                if value is None:
                    values.append("NULL")
                elif isinstance(value, str):
                    values.append(f"'{value.replace("'", "''")}'")
                else:
                    values.append(str(value))

            insert_statement = f"INSERT INTO {table_name} ({', '.join([f'`{col}`' for col in column_names])}) VALUES ({', '.join(values)});"
            cursor.execute(insert_statement)

        # Confirmar cambios
        conn.commit()

        # Cerrar conexión
        cursor.close()
        conn.close()

        print(f"Datos insertados correctamente en la tabla {table_name}.")

    except requests.exceptions.RequestException as e:
        print(f"Error al obtener datos de la API: {e}")
    except ValueError as e:
        print(f"Error en el formato de los datos: {e}")
    except mysql.connector.Error as e:
        print(f"Error al interactuar con la base de datos: {e}")

# Ejemplo de uso
if __name__ == "__main__":
    api_url = BASE_URL  # URL de la API
    table_name = "pscomppars"  # Nombre de la tabla de la base de datos

    # Configuración de la base de datos
    db_config = {
        'host': MYSQL_HOST,
        'user': MYSQL_USER,
        'password': MYSQL_PASS,
        'database': MYSQL_DB
    }

    json_to_inserts(api_url, table_name, db_config)
